src/daft-connect/src/session.rs

```diff
@@ -10,6 +11,9 @@ pub struct Session {
 
     id: String,
     server_side_session_id: String,
+    /// MicroPartitionSet associated with this session
+    /// this will be filled up as the user runs queries
+    pub(crate) pset: Arc<MicroPartitionSet>,
```

Comment:
so i could be abusing the `MicroPartitionSet`, but since `MicroPartitionSet` is already a batch of partitions `pub partitions: DashMap<PartitionId, Vec<Arc<MicroPartition>>>,`, it seemed redundant to have essentially `HashMap<String, HashMap<String, Vec<MicroPartition>>` this is what I was actually doing at first, but found there was **currently** no need for the outer hashmap. We'll likely need to refactor once we support distributed, but we're still a way away from that for spark.

Reply:
so is it usually

`GlobalHashMap<String, LocalHashMap<String, ...>>` in a distributed setting?